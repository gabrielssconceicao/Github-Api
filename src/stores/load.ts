import { ref } from 'vue';
import { defineStore } from 'pinia';

export const useLoadStore = defineStore('load', () => {
  const loading = ref(false);
  function setLoading() {
    loading.value = true;
  }
  function setNotLoading() {
    loading.value = false;
  }
  return { loading, setLoading, setNotLoading };
});
