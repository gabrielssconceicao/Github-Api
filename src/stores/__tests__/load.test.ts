import { describe, it, expect } from 'vitest';
import { useLoadStore } from '../load';

describe('loadStore', () => {
  it('should init with load = false', () => {
    const loadStore = useLoadStore();
    expect(loadStore.loading).toBe(false);
    expect({ loading: loadStore.loading }).toMatchSnapshot();
  });

  it('should change load to true', () => {
    const loadStore = useLoadStore();
    loadStore.setLoading();
    expect(loadStore.loading).toBe(true);
    expect({ loading: loadStore.loading }).toMatchSnapshot();
  });

  it('should change load to false', () => {
    const loadStore = useLoadStore();
    loadStore.setNotLoading();
    expect(loadStore.loading).toBe(false);
    expect({ loading: loadStore.loading }).toMatchSnapshot();
  });
});
