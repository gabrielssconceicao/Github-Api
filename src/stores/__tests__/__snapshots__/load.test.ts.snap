// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`loadStore > should change load to false 1`] = `
{
  "loading": false,
}
`;

exports[`loadStore > should change load to true 1`] = `
{
  "loading": true,
}
`;

exports[`loadStore > should init with load = false 1`] = `
{
  "loading": false,
}
`;
